Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionAzureNSG.yaml

```diff
@@ -0,0 +1,138 @@
+Parser:
+  Title: Network Session ASIM parser for Azure NSG logs
+  Version: '0.1'
+  LastUpdated: Feb 16, 2021
+Product:
+  Name: Azure NSG
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.2'
+References:
+  - Title: ASIM Network Session Schema
+    Link: https://aka.ms/ASimNetworkSessionDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing Azure NSG logs to the ASIM Network Session normalized schema.
+ParserName: ASimNetworkSessionAzureNSG
+EquivalentBuiltInParser: _ASim_NetworkSession_AzureNSG
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let DvcActionLookup = datatable(FlowStatus_s:string, DvcAction:string, EventResult:string) [
+      'A', 'Allow', 'Success',
+      'D', 'Deny', 'Failure',
+  ];
+  let NetworkDirectionLookup = datatable(FlowDirection_s:string, NetworkDirection:string, isOutBound:bool) [
+      'I', 'Inbound', false,
+      'O', 'Outbound', true
+  ];
+  let NetworkProtocolLookup = datatable(L4Protocol_s:string, NetworkProtocol:string)[
+      'T', 'TCP',
+      'U', 'UDP'
+  ];
+  let parser = (disabled:bool=false) 
+  {
+    let AzureNetworkAnalytics = 
+      AzureNetworkAnalytics_CL
+      | where not(disabled) and isnotempty(FlowType_s)
+      | lookup NetworkDirectionLookup on FlowDirection_s
```

Comment:
Th function is called twice, and lookup runs twice on both directions. If you first filter the  on flowdirection_s, and then lookup, it would perform better

Reply:
Good. Point. Updated. I woder if given that I anyway test for direction here, I should move all the contact part to here rather than the end. However, did not want to make big changes just for style.